ports/nrf/common-hal/digitalio/DigitalInOut.c

```diff
@@ -146,20 +140,21 @@ void common_hal_digitalio_digitalinout_set_pull(
             break;
     }
 
-    nrf_gpio_cfg_input(pin, hal_pull);
+    nrf_gpio_cfg_input(self->pin->number, hal_pull);
 }
 
 digitalio_pull_t common_hal_digitalio_digitalinout_get_pull(
         digitalio_digitalinout_obj_t *self) {
-    uint32_t pin = NRF_GPIO_PIN_MAP(self->pin->port, self->pin->pin);
+    uint32_t pin = self->pin->number;
+    // Changes pin to be a relative pin number in port.
     NRF_GPIO_Type *reg = nrf_gpio_pin_port_decode(&pin);
 
-    if (nrf_gpio_pin_dir_get(pin) == NRF_GPIO_PIN_DIR_OUTPUT) {
+    if (nrf_gpio_pin_dir_get(self->pin->number) == NRF_GPIO_PIN_DIR_OUTPUT) {
```

Comment:
To be consistent this should check self->output

Reply:
ditto